src/Ztm.WebApi/TransactionConfirmationWatchers/IRuleRepository.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using Ztm.WebApi.Callbacks;
+
+namespace Ztm.WebApi.TransactionConfirmationWatchers
+{
+    public interface IRuleRepository
+    {
+        Task<Rule> AddAsync
+        (
+            uint256 transaction,
+            int confirmation,
+            TimeSpan unconfirmedWaitingTime,
+            CallbackResult successResponse,
+            CallbackResult timeoutResponse,
+            Callback callback,
+            CancellationToken cancellationToken
+        );
+
+        Task ClearCurrentWatchIdAsync(Guid id, CancellationToken cancellationToken);
+        Task<Rule> GetAsync(Guid id, CancellationToken cancellationToken);
+        Task<TimeSpan> GetRemainingWaitingTimeAsync(Guid id, CancellationToken cancellationToken);
+        Task<RuleStatus> GetStatusAsync(Guid id, CancellationToken cancellationToken);
+        Task<IEnumerable<Rule>> ListWaitingAsync(CancellationToken cancellationToken);
+        Task SubtractRemainingWaitingTimeAsync(Guid id, TimeSpan remainingTime, CancellationToken cancellationToken);
```

Comment:
`remainingTime` looks confusing. My idea is just `value`.

Reply:
Changed it to `consumedTime`